test/common/EdgeConfiguration.cs

```diff
@@ -0,0 +1,216 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace common
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Collections.ObjectModel;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Newtonsoft.Json.Linq;
+
+    public enum UpstreamProtocolType
```

Comment:
Use Microsoft.Azure.Devices.Client.TransportType.

Reply:
The values of that enum don't line up with the strings expected in the deployment JSON. Wouldn't `Microsoft.Azure.Devices.Edge.Agent.Core.UpstreamProtocol` or `Microsoft.Azure.Devices.Edge.Hub.CloudProxy.UpstreamProtocol` be more appropriate?

I think I created my own enum in IotEdgeQuickstart because the available enums are buried (and duplicated) within the internals of Edge Agent and Edge Hub.

Reply:
yours are better :)